app/models/crime_application.rb

```diff
@@ -1,24 +1,33 @@
 class CrimeApplication < ApplicationRecord
   has_one :return_details, dependent: :destroy
 
-  attr_readonly :application, :submitted_at, :id
+  attr_readonly :submitted_details, :submitted_at, :id
   enum status: Types::ApplicationStatus.mapping
   enum review_status: Types::ReviewApplicationStatus.mapping
 
   before_validation :shift_payload_attributes, on: :create
+  before_validation :set_overall_offence_class, on: :create
 
   scope :by_status, ->(status) { where(status:) }
   scope :by_office, lambda { |office_code|
-    where("application->'provider_details'->>'office_code' = ?", office_code)
+    where("submitted_details->'provider_details'->>'office_code' = ?", office_code)
   }
 
   private
 
   def shift_payload_attributes
     return unless id.nil?
-    return unless application
+    return unless submitted_details
 
-    self.id = application.fetch('id')
-    self.submitted_at = application.fetch('submitted_at')
+    self.id = submitted_details.fetch('id')
+    self.submitted_at = submitted_details.fetch('submitted_at')
+  end
+
+  def set_overall_offence_class
+    return unless submitted_details
+
+    self.offence_class = Utils::OffenceClassCalculator.new(
```

Comment:
Perhaps wrap this in another method?

```ruby
def calculate_overall_offence_class
  Utils::OffenceClassCalculator.new(Styling with Markdown is supported
      offences: submitted_details['case_details']['offences']
    ).offence_class
end
```
Might make it easier to parse and also perhaps easier to change if we do.

So we just have

```ruby
 def set_overall_offence_class
    return unless submitted_details
    self.offence_class = calculate_overall_offence_class
end
```